dspace-api/src/main/java/org/dspace/content/ItemServiceImpl.java

```diff
@@ -1388,10 +1431,13 @@ public boolean canCreateNewVersion(Context context, Item item) throws SQLExcepti
         return resultingMetadataValueList;
     }
 
-    private MetadataValue getRelationMetadataFromOtherItem(Item otherItem, String relationName) {
-        MetadataValue metadataValue = constructMetadataValue(
-                MetadataSchemaEnum.RELATION.getName() + "." + relationName);
-        metadataValue.setAuthority("virtual");
+    private RelationshipMetadataValue getRelationMetadataFromOtherItem(Context context, Item otherItem,
+                                                                       String relationName,
+                                                                       Integer relationshipId) {
+        RelationshipMetadataValue metadataValue = constructMetadataValue(context,
+                                                                         MetadataSchemaEnum.RELATION
+                                                                             .getName() + "." + relationName);
+        metadataValue.setAuthority(Constants.VIRTUAL_AUTHORITY_PREFIX + relationshipId);
```

Comment:
Please consider that metadataValue can be null because constructMetadataValue might return null.

Reply:
Thanks @paulo-graca for identifying this. I've updated the branch with a check for null values from this method and some other small changes

Reply:
👍

Reply:
@benbosman I noticed the added verifications, but the one I pointed out is still unverified. The `metadataValue.setAuthority` on line 1440.

Reply:
@paulo-graca if I'm not mistaken, you're referring to https://github.com/DSpace/DSpace/blob/970b66e/dspace-api/src/main/java/org/dspace/content/ItemServiceImpl.java#L1458 in the current codebase?
This is verifying whether metadata value is null.

Reply:
My fault! Thank you! 👍